### PR TITLE
Refactor profile fetch code and add trainer features

### PR DESCRIPTION
**Pull request by nopparujp**

### CODE REVIEW

File: models/user_model.go [hunk 1]
```diff
@@ -12,10 +12,13 @@ import (
...
 
-func FindProfile(username string) (result map[string]interface{}, err error) {
+func FindProfile(username string, userType string) (result map[string]interface{}, err error) {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	filter := bson.D{{Key: "username", Value: username}}
+	if userType != "" {
+		filter = append(filter, bson.E{Key: "usertype", Value: userType})
+	}
 	opts := options.Find().SetProjection(bson.D{{"_id", 0}, {"hashedPassword", 0}, {"createdAt", 0}, {"updatedAt", 0}})
```
Comment (noppakorn) [inline models/user_model.go]: This part looks kinda weird to me. It seems that there is a FindOne function that you could use can could reduce a lot of code in this part. Please look in to that if it is not possible to use FindOne, then reply to this comment.

File: routes/trainer_routes.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"trainder-api/controllers"
+
+	"github.com/gin-gonic/gin"
+)
+
+func TrainerRoute(router *gin.Engine) {
+	router.GET("/trainer", controllers.GetTrainer())
+}
```
Comment (noppakorn) [inline routes/trainer_routes.go]: Make this a protected route also, only logged in user should be allow to find trainer right?

**Review by noppakorn — CHANGES_REQUESTED** [review]

**Comment by noppakorn** [comment]
For the trainer route, please also write the api comment. Then regenerate the swagger documentation. Read more about the docs generation process at the `README.md` of this repos.

File: controllers/user_controller.go [hunk 3]
```diff
@@ -109,3 +113,28 @@ func GetProfile() gin.HandlerFunc {
...
+				Status:  http.StatusBadRequest,
+				Message: `trainer profile retrieval unsuccessful`,
+			})
+			return
+		}
+		c.JSON(http.StatusFound, responses.GetProfileResponses{
+			Status:  http.StatusFound,
+			Message: `Successfully retrieve trainer profile`,
+			User:    result,
+		})
```
Comment (noppakorn) [inline controllers/user_controller.go]:
```suggestion
		c.JSON(http.StatusOK, responses.GetProfileResponses{
			Status:  http.StatusOK,
			Message: `Successfully retrieve trainer profile`,
			User:    result,
		})
```

Comment (noppakorn) [reply controllers/user_controller.go]: `StatusFound` is only used for redirect. You need to always return StatusOK for successful request.

Comment (nopparujp) [reply controllers/user_controller.go]: roger that

Comment (noppakorn) [reply controllers/user_controller.go]: Merging is done incorrectly please change to StatusOK

**Review by noppakorn — APPROVED** [review]
LGTM!

**Pull request merged**